pkg/healthcheck/healthcheck.go

```diff
@@ -1361,6 +1374,32 @@ func (hc *HealthChecker) checkCanGet(namespace, group, version, resource string)
 	return hc.checkCanPerformAction("get", namespace, group, version, resource)
 }
 
+func (hc *HealthChecker) checkAPIService(serviceName string) error {
+	apiServiceClient, err := apiregistrationv1client.NewForConfig(hc.kubeAPI.Config)
+	if err != nil {
+		return err
+	}
+
+	apiStatus, err := apiServiceClient.APIServices().Get(serviceName, metav1.GetOptions{})
+	if err != nil && len(apiStatus.Status.Conditions) == 0 {
+		return err
+	}
+
+	available := false
+	var errorMessage string
+	for _, condition := range apiStatus.Status.Conditions {
+		if condition.Status == "True" {
+			available = true
+			break
+		}
+		errorMessage = fmt.Sprintf("%s: %s", condition.Reason, condition.Message)
```

Comment:
@droidnoob looking at this again, I am wondering whether this logic is very robust. So as per the comments in the k8s api this `Conditions` list is a list of objects: `APIServiceCondition describes the state of an APIService at a particular point`.  Here you are assuming that if you see a condition with status `True` its all good. But is it? What if there was a condition right after it that was generated at a later point in time and was with status `Unknown`. Moreover, is the order of these statuses in the list guaranteed to be time dependent? 

I think the more sane way to do is to get the condition with the highest `LastTransitionTime` and check that one. @adleong @Pothulapati Does that make sense. I have not spent too much time with the k8s API

Reply:
@zaharidichev I was following what @adleong suggested me to do. https://github.com/linkerd/linkerd2/pull/3689#discussion_r344300551

@adleong Should there be changes like @zaharidichev mentioned?

Reply:
my understanding of this object is that there may be a list of condition objects.  each condition has a type and and status.  I think we should be checking that the condition with type "Available" has status == "True".  In other words, we should iterate through the list of conditions until we find the `Type == Available` condition and check that that condition has `Status == "True"`. 

Does that make sense?

Reply:
Yes it makes perfect sense. I was thinking that this is simply a list of conditions that have been transited to ordered in a chronological order. But I am wrong I guess